### PR TITLE
Fix for 32bit Julia

### PR DESCRIPTION
**Pull request by bradcarman**

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -223,22 +223,28 @@ Base.nameof(s::BasicSymbolic) = issym(s) ? s.name : error("None Sym BasicSymboli
 
 ## This is much faster than hash of an array of Any
 hashvec(xs, z) = foldr(hash, xs, init=z)
+
+const csym = typemax(UInt)
```
Comment (YingboMa) [inline src/types.jl:227]: It should be something like `const SYM_SALT = 0x4de7d7c66d41da43 % UInt`

Comment (bradcarman) [reply src/types.jl:227]: But this is a UInt64.  hash(::Symbol, ::UInt64) isn't supported by 32bit Julia (as best I can tell)

Comment (YingboMa) [reply src/types.jl:227]: `0x4de7d7c66d41da43 % UInt` is UInt

Comment (bradcarman) [reply src/types.jl:227]: In Julia 32bit, this is what I get...

```julia
julia> const SYM_SALT = 0x4de7d7c66d41da43
0x4de7d7c66d41da43

julia> typeof(SYM_SALT)
UInt64

julia> hash(:symbol, SYM_SALT)
ERROR: MethodError: no method matching hash(::Symbol, ::UInt64)
```

Comment (YingboMa) [reply src/types.jl:227]: That's not `0x4de7d7c66d41da43 % UInt` though.

Comment (bradcarman) [reply src/types.jl]: OK, got it, sorry I kept seeing something different 😅 .  Fixed now.

**Pull request merged**